wrapper/src/main/java/software/amazon/jdbc/plugin/AuroraConnectionTrackerPlugin.java

```diff
@@ -150,8 +173,7 @@ public void notifyNodeListChanged(final Map<String, EnumSet<NodeChangeOptions>>
   }
 
   private boolean isRoleChanged(final EnumSet<NodeChangeOptions> changes) {
-    return changes.contains(NodeChangeOptions.PROMOTED_TO_WRITER)
-        || changes.contains(NodeChangeOptions.PROMOTED_TO_READER);
+    return changes.contains(NodeChangeOptions.PROMOTED_TO_READER);
   }
 
   public String getInstanceEndpoint(final Connection conn, final HostSpec host) {
```

Comment:
It should return null id dialect doesn't support getting an instance name. For now it returns "?" that makes no sense.